### PR TITLE
Add Playwright CT tests for TradeboxMarginFields

### PR DESCRIPTION
**Pull request by qwinndev**
## Summary
- Adds Playwright Component Tests and Vitest hook tests for `TradeboxMarginFields` (FEDEV-3678)
- Covers field recalculation, anchor behavior, price change reactions, display modes, slider sync, and edge cases
- Sets up CT infrastructure (playwright-ct.config, global styles, mock modules)
- Fixes CT build OOM via `NODE_OPTIONS=--max-old-space-size=8192`

### Test breakdown

**Playwright CT (103 tests in 4 files)**

| File | Tests | Covers |
|---|---|---|
| `TradeboxMarginFields.ct.spec` | 24 | Rendering, inputs, display mode toggle, price field, slider |
| `TradeboxMarginFieldsIntegration.ct.spec` | 50 | Margin/size recalculation, price change sync, anchor behavior, slider ↔ value, ETH margin (pay ≠ collateral), leverage OFF slider sync |
| `TradeboxMarginFieldsEdgeCases.ct.spec` | 14 | Zero/empty values, undefined tokens, rapid input, large/dust amounts |
| `Tabs.ct.spec` | 15 | Tabs component rendering, interaction, QA attributes |

**Vitest (22 tests in 2 files)**

| File | Tests | Covers |
|---|---|---|
| `useSizeConversion.spec` | 12 | tokensToUsd, usdToTokens, edge cases |
| `useTradeboxManualLeverageSizeSlider.spec` | 10 | sizePercentage, handleSizePercentageChange, markFieldInteraction |

## Test plan
- [x] All 103 Playwright CT tests pass locally
- [x] All 22 Vitest hook tests pass locally
- [x] ESLint passes with zero warnings
- [x] Pre-push hook (vitest + playwright CT) passes

### CODE REVIEW

File: src/components/TradeboxMarginFields/__tests__/testFixtures.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+import type { SyntheticsState } from "context/SyntheticsStateContext/SyntheticsStateContextProvider";
+import { expandDecimals } from "lib/numbers";
+import type { DeepPartial } from "lib/types";
+import type { TokenData } from "sdk/utils/tokens/types";
+import { TradeMode, TradeType } from "sdk/utils/trade/types";
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+export const noop = () => {};
+
+export const USDC_ADDRESS = "0xaf88d065e77c8cC2239327C5EDb3A432268e5831";
```
Comment (divhead) [inline src/components/TradeboxMarginFields/__tests__/testFixtures.ts]: hmm, can we use token addresses from configs?

File: src/components/TradeboxMarginFields/__tests__/testFixtures.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  triggerPriceInputValue?: string;
+  isLeverageSliderEnabled?: boolean;
+  fromTokenAddress?: string;
+  toTokenAddress?: string;
+  marketAddress?: string;
+  tokensData?: Record<string, TokenData>;
+  marketInfo?: { marketTokenAddress: string; [key: string]: unknown };
+};
+
+export function createMockState(overrides: MockStateOverrides = {}): SyntheticsState {
```
Comment (divhead) [inline src/components/TradeboxMarginFields/__tests__/testFixtures.ts]: let's move potential global mocks to global context? like domain/synthetics/__tests__

File: src/components/TradeboxMarginFields/__tests__/useTradeboxManualLeverageSizeSlider.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+import type { DeepPartial } from "lib/types";
+import type { MarketInfo } from "sdk/utils/markets/types";
+import type { TokenData } from "sdk/utils/tokens/types";
+import { TradeMode, TradeType } from "sdk/utils/trade/types";
+
+import { useTradeboxManualLeverageSizeSlider } from "../useTradeboxManualLeverageSizeSlider";
+
+const USDC_ADDRESS = "0xaf88d065e77c8cC2239327C5EDb3A432268e5831";
+const ETH_ADDRESS = "0x82aF49447D8a07e3bd95BD0d56f35241523fBab1";
+const MARKET_ADDRESS = "0x70d95587d40A2caf56bd97485aB3Eec10Bee6336";
```
Comment (divhead) [inline src/components/TradeboxMarginFields/__tests__/useTradeboxManualLeverageSizeSlider.spec.tsx]: here also, can we use addresses from config?

File: src/components/TradeboxMarginFields/__tests__/utils.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export function getDataQALocator(name: string): string {
```
Comment (divhead) [inline src/components/TradeboxMarginFields/__tests__/utils.ts]: this is also looks like a global util

Comment (qwinndev) [reply src/components/TradeboxMarginFields/__tests__/utils.ts]: fixed

File: src/components/TradeboxMarginFields/__tests__/useTradeboxManualLeverageSizeSlider.spec.tsx (hunk 3 above)
Comment (qwinndev) [reply src/components/TradeboxMarginFields/__tests__/useTradeboxManualLeverageSizeSlider.spec.tsx]: fixed

File: src/components/TradeboxMarginFields/__tests__/testFixtures.ts (hunk 2 above)
Comment (qwinndev) [reply src/components/TradeboxMarginFields/__tests__/testFixtures.ts]: fixed

File: src/components/TradeboxMarginFields/__tests__/testFixtures.ts (hunk 1 above)
Comment (qwinndev) [reply src/components/TradeboxMarginFields/__tests__/testFixtures.ts]: fixed

File: src/components/TradeboxMarginFields/__tests__/useSizeConversion.spec.ts [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+}
+
+function usdToTokens(usdValue: string, decimals: number, markPrice: bigint, visualMultiplier?: number): string {
+  if (markPrice === 0n) return "";
+  const parsed = parseValue(usdValue || "0", USD_DECIMALS);
+  if (parsed === undefined) return "";
+  const tokens = convertToTokenAmount(parsed, decimals, markPrice);
+  if (tokens === undefined) return "";
+  return formatAmountFree(tokens, decimals, TOKEN_INPUT_DISPLAY_DECIMALS, visualMultiplier);
+}
```
[the remark this answers is not in the document]
Comment (qwinndev) [reply src/components/TradeboxMarginFields/__tests__/useSizeConversion.spec.ts]: fixed
